tests/test_publish_dafsa.py

```diff
@@ -0,0 +1,148 @@
+            get_latest_hash("".join(COMMIT_HASH_URL.split("repo")))
+
+    def test_download_resources(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            download_resources(tmp, PREPARE_TLDS_PY, MAKE_DAFSA_PY, LIST_URL)
+            self.assertEqual(
+                sorted(os.listdir(tmp)),
+                sorted(["public_suffix_list.dat", "prepare_tlds.py", "make_dafsa.py"]),
+            )
+            with self.assertRaises(requests.exceptions.HTTPError):
+                download_resources(tmp, PREPARE_TLDS_PY + "c"),
+
+
+class TestPrepareDafsa(unittest.TestCase):
+    def test_prepare_dafsa(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            output_binary_path = prepare_dafsa(tmp)
+            self.assertIn(os.path.basename(output_binary_path), os.listdir(tmp))
+            self.assertGreater(os.path.getsize(output_binary_path), 0)
+
+    def test_exception_is_raise_when_process_returns_non_zero(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            with mock.patch("subprocess.Popen") as mocked:
+                mocked.return_value.returncode = 1
+                with self.assertRaises(Exception) as cm:
+                    prepare_dafsa(tmp)
+                    self.assertIn("DAFSA Build Failed", str(cm.exception))
+
+
+class TestRemoteSettingsPublish(unittest.TestCase):
+    @responses.activate
+    def test_remote_settings_publish(self):
+        server = "https://kinto.dev.mozaws.net/v1"
+        record_uri = f"{server}/buckets/main-workspace/collections/public-suffix-list/records/tld-dafsa"  # noqa
+        attachment_uri = f"{record_uri}/attachment"
+
+        responses.add(
+            responses.PUT,
+            f"{server}/accounts/arpit73",
+            adding_headers={"Content-Type": "application/json"},
+            json='{"data": {"password": "pAsSwErD"}}',
+        )
+        client = Client(
+            server_url=server,
+            auth=("arpit73", "pAsSwErD"),
+            bucket=BUCKET_ID,
+            collection=COLLECTION_ID,
+        )
+        self.assertEqual(
+            client.get_endpoint("record", id=RECORD_ID),
+            "/buckets/dafsa-bucket/collections/public-suffix-list/records/tld-dafsa",
+        )
+
+        responses.add(
+            responses.POST,
+            attachment_uri,
+            adding_headers={"Content-Type": "multipart/form-data"},
+            files={"attachment": ("dafsa.bin", b"some binary data")},
+            json='{"commit_hash": "abc"}',
+        )
+        responses.add(
+            responses.PATCH,
+            record_uri,
+            adding_headers={"Content-Type": "application/json"},
+            json='{"data": {"status": "to-review"}}',
+        )
+
+        with tempfile.TemporaryDirectory() as tmp:
+            with open(f"{tmp}/dafsa.bin", "wb") as f:
+                f.write(b"some binary data")
+            remote_settings_publish(client, "abc", f"{tmp}/dafsa.bin")
+
```

Comment:
Ok, so now that you intercepted the HTTP requests with the `responses` package, and that you executed the publish function, you could assert that the POST was sent.

```py
self.assertEqual(len(responses.calls), 2)
self.assertEqual(responses.calls[0].request.method, "post")
...
```